models/model_choice.py

```diff
@@ -136,12 +135,11 @@ def set_hyperparameters(params,
     gamma = get_key_def('gamma', params['scheduler']['params'], 0.9)
     class_weights = torch.tensor(class_weights) if class_weights else None
     # Loss function
-    if num_classes == 1:
-        criterion = SingleClassCriterion(loss_type=loss_fn, ignore_index=dontcare_val)
+    if loss_fn['_target_'] in ['torch.nn.CrossEntropyLoss', 'losses.focal_loss.FocalLoss',
```

Comment:
We lose the differentiator between SingleClass and MultiClass here, if a user were to pass num_classes == 1 while picking CE loss he will almost certainly run into errors at some point in the code execution

Reply:
Good point. I can add a small check for this.